tools/federation/src/main/java/org/eclipse/rdf4j/federated/FederationManager.java

```diff
@@ -85,124 +72,27 @@
-		FedX federation = new FedX(members);
-
-		FedXRepository repo = new FedXRepository(federation);
-
-		instance = new FederationManager(federation, cache, ex, repo);
-		instance.updateStrategy();
-		instance.reset();
-
-		try {
-			repo.init();
-		} catch (RepositoryException e) {
-			// should never occur
-			throw new FedXRuntimeException(e);
-		}
-
-		EndpointManager.initialize(members);
-
-		if (Config.getConfig().isEnableJMX()) {
-			try {
-				MonitoringUtil.initializeJMXMonitoring();
-			} catch (Exception e1) {
-				log.error("JMX monitoring could not be initialized: " + e1.getMessage());
-			}
-		}
-
-		// initialize prefix declarations, if any
-		String prefixFile = Config.getConfig().getPrefixDeclarations();
-		if (prefixFile != null) {
-			QueryManager qm = instance.getQueryManager();
-			Properties props = new Properties();
-			try (FileInputStream fin = new FileInputStream(new File(prefixFile))) {
-				props.load(fin);
-			} catch (IOException e) {
-				throw new FedXRuntimeException("Error loading prefix properties: " + e.getMessage());
-			}
-
-			for (String ns : props.stringPropertyNames()) {
-				qm.addPrefixDeclaration(ns, props.getProperty(ns)); // register namespace/prefix pair
-			}
-		}
-
-		return repo;
-	}
-
-	/**
-	 * Return the initialized {@link FederationManager} instance.
-	 * 
-	 * @return the federation manager
-	 */
-	public static FederationManager getInstance() {
-		if (instance == null)
-			throw new FedXRuntimeException("FederationManager has not been initialized yet, call #initialize() first.");
-		return instance;
-	}
-
-	/**
-	 * Returns true if the {@link FederationManager} is initialized.
-	 * 
-	 * @return true or false;
-	 */
-	public static boolean isInitialized() {
-		return instance != null;
-	}
-
-	static Monitoring monitoring;
-
-	public static Monitoring getMonitoringService() {
-		if (!isInitialized())
-			throw new IllegalStateException("Monitoring service can only be used if FedX is initialized.");
-		return monitoring;
-	}
-
 	/* Instance variables */
+	protected FederationContext federationContext;
 	protected FedX federation;
-	protected Cache cache;
 	protected ExecutorService executor;
 	protected FederationEvalStrategy strategy;
 	protected FederationType type;
 	protected ControlledWorkerScheduler<BindingSet> joinScheduler;
 	protected ControlledWorkerScheduler<BindingSet> leftJoinScheduler;
 	protected ControlledWorkerScheduler<BindingSet> unionScheduler;
 
-	private FederationManager(FedX federation, Cache cache, ExecutorService executor,
-			Repository repo) {
-		this.federation = federation;
-		this.cache = cache;
-		this.executor = executor;
-		QueryManager.instance = new QueryManager(this, repo); // initialize the singleton query manager
+	public FederationManager() {
+
 	}
 
-	public FedX getFederation() {
-		return federation;
+	public void initialize(FedX federation, FederationContext federationContext) {
```

Comment:
Same here - for the Sail, the Repository, and the RepositoryManager, we renamed `initialize` to `init`. See #552 for details.

Reply:
Makes sense to align